articles/exception-handling-in-javascript/index.md

```diff
@@ -0,0 +1,159 @@
+
+### Prerequisites
+Exception Handling in JavaScript is an exciting and interactive topic. Therefore, I recommend the reader to have a basic understanding of HTML, CSS, and JavaScript.
+
+### What is Exception Handling
+Exception handling is one of the powerful JavaScript features to handle errors and maintain a regular JavaScript code/program flow. An exception is an object with a description of what went wrong. Also, it discovers where the problem occurred.
+Errors occur due to mistakes made by developers, wrong input, or unforeseeable things. Some reasons why exceptions occur are:
+
+- Dividing a number by zero: This results into infinity, thus throwing an exception.
+- When a file that is requested does not exist in the system.
+- When the user provides the wrong input.
+- When the network drops during communication.
+
+If a software engineer fails to plan for failure, then whatever project or code they are working on may not be successful. That is where [exception handling](https://en.wikipedia.org/wiki/Exception_handling) comes in. When JavaScript encounters an error and raises an exception. The JavaScript translator looks for an exception handling code instead of proceeding to the next statement. In a programming environment, exceptions can be handled, but errors cannot.
+
+### JavaScript Error Types
+Different errors may occur while executing a JavaScript code. There are three types of errors.
+
+1. [**Syntax Errors:**](https://developer.mozilla.org/en-US/docs/Glossary/Syntax_error#:~:text=An%20exception%20caused%20by%20the,you%20trigger%20a%20syntax%20error.) errors that cannot be interpreted by the computer. These errors stop the program from working. In JavaScript, these errors are:
+
+- Spelling errors(wrong spelling such as fiction instead of function)
+- The omission of important characters such as not using a semicolon to end a statement
+- Use of the wrong indentation
+
+2. [**Runtime Errors:**](https://www.geeksforgeeks.org/javascript-error-and-exceptional-handling-with-examples/#:~:text=Runtime%20Error%3A%20A%20runtime%20error,also%20known%20as%20the%20exceptions.&amp;text=try%20___%20catch%20method%3A%20JavaScript,operator%20to%20handle%20the%20exception.)these errors take place during execution. The errors are detected when your program runs. It crashes or raises an exception. Thus, exception handlers handle exception errors. These errors are often caused by:
+
+- The program cannot find data because it does not exist.
+- It cannot act on the data because it is an invalid type of data.
+
+3. [**Logical Errors:**](https://study.com/academy/lesson/errors-in-javascript-types-methods-prevention.html) these are errors that take place due to logical mistakes. These types of errors do not throw an error or an exception at all. But do not do what the developer intends the code to do. It is challenging to find logical errors. They can only be found through extensive testing.
```

Comment:
You can find a better explanation for this type of error. It sounds a bit vague as it is. You could add something like

`these types of errors do not throw an error or an exception at all. This is because they result from the code not doing what the developer intends it to. It is challenging..."